src/zjs_timers.c

```diff
@@ -162,7 +166,6 @@ static bool delete_timer(zjs_timer_t *tm)
 #endif
             zjs_remove_callback(tm->callback_id);
         }
-        ZJS_LIST_REMOVE(zjs_timer_t, zjs_timers, tm);
```

Comment:
Moving this worries me because this code has been modified and debugged a lot. Probably similar to my comment below, I expect the real problem here was that delete_timer was called when it shouldn't be. What was that situation?

We should only call `delete_timer` when we know there is a timer to delete and we are going to delete it. It shouldn't be "delete timer unless it isn't there after all and that's just fine".